Source/missiles.cpp

```diff
@@ -936,12 +930,15 @@ bool Plr2PlrMHit(int pnum, int p, int mindam, int maxdam, int dist, int mtype, b
 	return false;
 }
 
-void CheckMissileCol(int i, int mindam, int maxdam, bool shift, int mx, int my, bool nodel)
+void CheckMissileCol(int i, int mindam, int maxdam, bool shift, Point position, bool nodel)
 {
 	int oi;
 	bool blocked;
 	int dir, mAnimFAmt;
 
+	int mx = position.x;
+	int my = position.y;
+
 	if (i >= MAXMISSILES || i < 0)
 		return;
 	if (mx >= MAXDUNX || mx < 0)

```

Comment:
The checks on `mx, my` could potentially use `!InDungeonBounds()`, if it weren't for the <0 check which doesn't quite match...